### PR TITLE
=clt #18314 Reduce message lost when region terminates

### PR DESCRIPTION
**Pull request by patriknw**
The new `akka.cluster.down-removal-margin` comes into play.
During that period messages are still routed to the old location, even though we have got the Terminated message.

We can reduce (best effort) the message loss by not replying to GetShardHome during the period.

### CODE REVIEW

**Comment by patriknw** [comment]
I think that failure was the same as was fixed in other PR. Will rebase.

**Comment by ktoso** [comment]
Great improvement, LGTM

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala [hunk 1]
```diff
@@ -564,10 +570,10 @@ abstract class ShardCoordinator(typeName: String, settings: ClusterShardingSetti
       log.debug("ShardRegion terminated: [{}]", ref)
       state.regions(ref).foreach { s ⇒ self ! GetShardHome(s) }
 
-      gracefulShutdownInProgress -= ref
```
Comment (drewhk) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: what is the reason for this moved into the block below?

Comment (patriknw) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: There is no difference in practise, but it feels (reads) better to modify the state of the actor after persist of ShardRegionTerminated has been done. The reason it doesn't matter is that no other message is processed in-between (stashing) and if the persist fails the actor will be stopped anyway.

Comment (drewhk) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: Ok, thanks!

**Comment by drewhk** [comment]
LGTM

**Comment by patriknw** [comment]
Refs #18314

**Pull request merged**